### PR TITLE
Feat/imgur2 s3

### LINKED ISSUE #35956: Replace monorepo imgur images with S3
node_modules, .cache, public directories not included

Corner cases: 
1. there could be more than one instance of each imge
2. some urls end in \ in the codebase

3. some urls are in the tests ```(https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)```

4. some urls do not end in photo format.
https://imgur.com/Yl0Bxwn
https://imgur.com/wgscRNa
5. some photos have dimensions
https://i.stack.imgur.com/jRaTj.jpg?s=128&g=1
6. some links are http 
http://i.imgur.com/Z1DSMYC.png

here are the unique urls in the codebase:

['https://i.imgur.com/U1CyEuA.jpg',
'https://i.imgur.com/HBVrdaj.jpg',
'https://i.imgur.com/701RLfV.jpg',
'https://i.imgur.com/tKUi6Rf.jpg',
'https://i.imgur.com/op8BVph.jpg',
'https://i.imgur.com/G2YvPHf.jpg',
'https://i.imgur.com/nsvNixW.jpg',
'https://i.imgur.com/QPpjPac.jpg',
'https://i.imgur.com/wjlDigg.jpg',
'https://i.imgur.com/MJAkxbh.png',
'https://i.imgur.com/1Atixoj.gif',
'https://i.imgur.com/eWWi3gZ.gif',
'https://i.imgur.com/aa8oGgP.png',
'https://i.stack.imgur.com/4bz2N.png',
'https://i.stack.imgur.com/5LZ9V.png',
'https://i.imgur.com/QAuugOL.jpg',
'https://i.imgur.com/1710fEx.jpg',
'https://i.imgur.com/jr4icze.jpg',
'https://i.imgur.com/W57lGsk.jpg',
'https://i.imgur.com/5Twm8NE.jpg',
'https://i.imgur.com/01xetay.jpg',
'https://i.imgur.com/ThkzYEV.jpg',
'https://i.imgur.com/K7HfCly.jpg',
'https://i.imgur.com/ZOSxeZD.jpg',
'https://i.imgur.com/SwKsPxD.jpg',
'https://i.imgur.com/3AaViff.jpg',
'https://i.imgur.com/RMTqXPw.png',
'https://i.stack.imgur.com/59Rpw.png',
'https://i.imgur.com/FwZuFGy.png',
'https://i.imgur.com/d1liGwI.png',
'https://imgur.com/SiU8uTe.png',
'https://i.imgur.com/Pb0aTkV.png',
'https://i.imgur.com/YdbgWHL.png',
'https://i.imgur.com/2uGfleD.jpg',
'https://i.imgur.com/CvtXCMd.jpg',
'https://i.imgur.com/e4dRvZR.jpg',
'https://i.imgur.com/k4FSyXz.jpg',
'https://i.imgur.com/YvtbUBH.jpg',
'https://i.imgur.com/8gKRhAp.jpg',
'https://i.imgur.com/lFJaBUfh.png',
'https://i.imgur.com/4X8ITZih.png',
'https://i.imgur.com/wPqUgsih.png',
'https://i.imgur.com/7n0V1G4h.png',
'https://i.imgur.com/pa4grM8h.png',
'https://i.imgur.com/6Z2VStD.png',
'https://imgur.com/sczZjwm.gif',
'https://i.imgur.com/EkjwJdr.png',
'https://i.imgur.com/zoMspyn.png',
'https://i.imgur.com/3d9SqBR.png',
'https://i.imgur.com/rnQEY7o.png',
'https://i.imgur.com/jrnu6mE.png',
'https://i.imgur.com/gszPtRa.png',
'https://i.imgur.com/TKXxZGF.png',
'https://i.imgur.com/qQGOiyC.png',
'https://i.imgur.com/uDbypks.png',
'https://i.imgur.com/u7dfnuA.png',
'https://i.imgur.com/6x8Z0pj.png',
'https://i.imgur.com/8EsVvZk.png',
'https://i.imgur.com/V2JzAka.png',
'https://i.imgur.com/KUeOSK3.png',
'https://i.imgur.com/SOhv2jZ.png',
'https://i.stack.imgur.com/gCaz3.png',
'https://i.imgur.com/YXzUPiB.png',
'https://i.imgur.com/AQ3P4vi.png',
'https://i.imgur.com/ZErXKtQ.png',
'https://i.imgur.com/fN5wOH2.png',
'https://i.imgur.com/4ji7JXx.png',
'https://i.imgur.com/1tPHKHj.png',
'https://i.imgur.com/l8EAw1L.png',
'https://i.imgur.com/pfC7QFH.png',
'https://i.imgur.com/b4vFsKa.png',
'https://i.imgur.com/uRLunzU.png',
'https://i.imgur.com/pZij95O.jpg',
'https://i.imgur.com/vlV7ZFr.png',
'https://i.imgur.com/9ILjA1q.jpg',
'https://i.imgur.com/V5nP3xj.jpg',
'https://i.imgur.com/DznuAou.jpg',
'https://i.imgur.com/O16uc8E.png',
'https://i.imgur.com/AyvSsLk.png',
'https://i.imgur.com/4RY0hVu.png',
'https://i.imgur.com/hsEjqfm.png',
'https://i.imgur.com/RqgGcf5.gif',
'https://i.imgur.com/msFhj0x.png',
'https://i.imgur.com/ZI8P3Qx.png',
'https://i.imgur.com/W6QaUAX.png',
'https://i.imgur.com/1UUFXnM.png',
'https://i.imgur.com/Lih2m92.png',
'https://i.imgur.com/mEvXx3A.png',
'https://i.imgur.com/tYzAJK1.png',
'https://i.imgur.com/oQc1SKK.png',
'https://i.imgur.com/daIlXQI.png',
'https://i.imgur.com/S0QPDi4.png',
'https://i.imgur.com/Yl0Bxwn.gif',
'https://i.imgur.com/wgscRNa.gif',
'https://i.imgur.com/ToWQjHO.png',
'https://i.imgur.com/eULdyep.png',
'https://i.imgur.com/r0br52L.png',
'https://i.stack.imgur.com/ALoC9.png',
'https://i.imgur.com/W3amiIr.png',
'https://i.imgur.com/5ys9G3k.png',
'https://i.imgur.com/Jq0eKrO.jpg',
'https://i.imgur.com/3ALtws1.png',
'https://i.imgur.com/Jk2aJod.png',
'https://i.imgur.com/fassWYs.png',
'https://i.imgur.com/H9mO8qS.jpg',
'https://i.imgur.com/woj5vPm.png',
'https://i.imgur.com/halOgkj.png',
'https://i.imgur.com/Z1DSMYC.png',
'https://i.stack.imgur.com/jRaTj.jpg']

### PR DESCRIPTION
**Pull request by ahmaxed**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [ ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

closes #35956

imgur urls in node_modules, .cache, public directories are not included

Corner cases:

- [ ] there could be more than one instance of each imge

- [x]  some urls end in \ in the codebase

- [x] some urls are in the tests (https:\/\/i\.imgur\.com\/MJAkxbh\.png\1\s*\)

- [x] some urls do not end in photo format.
https://imgur.com/Yl0Bxwn
https://imgur.com/wgscRNa

- [x] some photos have dimensions
https://i.stack.imgur.com/jRaTj.jpg?s=128&g=1

- [x] some links are http
http://i.imgur.com/Z1DSMYC.png

### CODE REVIEW

File: guide/arabic/miscellaneous/storing-mongo-username--password-persistently-using-dotenv/index.md [hunk 1]
```diff
@@ -69,7 +68,7 @@ node\_modules
 
 [stackoverflow.com](http://stackoverflow.com/questions/35356692/best-practice-when-using-an-api-key-in-node-js)
 
-[! \[Drake Main](https://i.stack.imgur.com/jRaTj.jpg?s=128&g=1) \] (http://stackoverflow.com/users/4956243/drake-main) #### [أفضل الممارسات عند استخدام مفتاح API في Node.js](http://stackoverflow.com/questions/35356692/best-practice-when-using-an-api-key-in-node-js)
+[! \[Drake Main](https://s3.amazonaws.com/cdn-media-1.freecodecamp.org/imgr/jRaTj.jpg) \] (http://stackoverflow.com/users/4956243/drake-main) #### [أفضل الممارسات عند استخدام مفتاح API في Node.js](http://stackoverflow.com/questions/35356692/best-practice-when-using-an-api-key-in-node-js)
```
Comment (raisedadead) [inline guide/arabic/miscellaneous/storing-mongo-username--password-persistently-using-dotenv/index.md]: This is an incorrect format for the URL. 
```suggestion
[! \[Drake Main](https://cdn-media-1.freecodecamp.org/imgr/jRaTj.jpg) \] (http://stackoverflow.com/users/4956243/drake-main) #### [أفضل الممارسات عند استخدام مفتاح API في Node.js](http://stackoverflow.com/questions/35356692/best-practice-when-using-an-api-key-in-node-js)
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
Incorrect format for the CDN URL.

**Review by raisedadead — CHANGES_REQUESTED** [review]
Also make sure, you make no additional changes (including formatting) than swapping the image URLs.

**Comment by raisedadead** [comment]
Just one more thought, you probably want to make lots of smaller PRs changing just a few files. This way your PR will not be held up due to merge conflicts.

/cc @freeCodeCamp/moderators

**Comment by ahmadabdolsaheb** [comment]
@raisedadead I think I will be able to complete the whole thing today. Would I still need to make small prs?

**Comment by moT01** [comment]
If you search for your PR# [here](https://contribute.freecodecamp.org/dashboard/) - you can see what files would potentially create conflicts. You could maybe just remove those files from this PR and put them on a different one.

**Comment by thecodingaviator** [comment]
> @raisedadead I think I will be able to complete the whole thing today. Would I still need to make small prs?

@ahmadabdolsaheb Yes please, that'll make it a lot easier

**Comment by ahmadabdolsaheb** [comment]
@moT01, @thecodingaviator, the pr is divided in 8 prs. one pr is for files with potential conflicts. :)

**Comment by RandellDawson** [comment]
@ahmadabdolsaheb @raisedadead I have reviewed all 8 of the PRs referenced above and made suggested comments where applicable.  

The one question I have (even though it is probably bike-shedding) is if these are `imgur` images, shouldn't the base url be:
`"https://cdn-media-1.freecodecamp.org/imgur`
instead of the currently used url:
`"https://cdn-media-1.freecodecamp.org/imgr`

**Comment by ahmadabdolsaheb** [comment]
@RandellDawson Thanks you for your review.  
That is a relevant question and shows your attention to details.

That was an attempt to make the url one letter shorter and still make the source recognizable. 

However, it would not matter a lot since all of them will be replaced with an environmental variable in the near future.

**Comment by RandellDawson** [comment]
@ahmadabdolsaheb Did you mean to close this PR?
